app/process/handler/consensus-message/consensus-message.go

```diff
@@ -124,84 +124,84 @@ func (cmh ConsensusMessageHandler) handlePayload(payload []byte) error {
 	}
 
-	requiredSigCount := len(cmh.validAddresses)/2 + len(cmh.validAddresses)%2
-	log.Infof("Required signatures: [%v]", requiredSigCount)
-
-	txSignatures, err = cmh.repository.GetByTransactionId(m.TransactionId, hexHash)
-	if err != nil {
-		return errors.New(fmt.Sprintf("Could not retrieve Transaction Signatures for Transaction [%s]", m.TransactionId))
-	}
-
-	if len(txSignatures) < requiredSigCount {
-		log.Warnf(fmt.Sprintf("Insignificant amount of Transaction Signatures for Transaction [%s] - [%d] signatures", m.TransactionId, len(txSignatures)))
-		return nil
+	if cmh.enoughSignaturesCollected(txSignatures, m.TransactionId) {
+		log.Infof("Signatures for TX ID [%s] were collected. Proceeding with leader election.", m.TransactionId)
+		go cmh.start(txSignatures, hexHash)
+		log.Infof("Started leader election and preparation of aggregated signatures message.")
 	}
+	return nil
+}
 
-	pos := cmh.findMyPosition(txSignatures)
-	if pos == -1 {
-		return errors.New(fmt.Sprintf("Operator is not amongst the potential leaders - [%v]", pos))
+func (cmh ConsensusMessageHandler) start(txSignatures []message.TransactionMessage, hash string) {
+	pos, err := cmh.findMyPosition(txSignatures)
+	if err != nil {
+		log.Errorf("Failed in finding leader position: [%s]", err)
+		return
 	}
 
-	log.Infof("My position: [%v]", pos)
-
 	now := time.Now()
-	deadline := now.Add(time.Minute * time.Duration(cmh.deadline*pos))
-
-	log.Infof("NOW: [%v]", now)
-	log.Infof("DEADLINE: [%v]", deadline)
+	deadline := now.Add(time.Second * time.Duration(cmh.deadline*pos))
 
 	for now.Before(deadline) {
 		if cmh.transactionSent() {
-			return nil
+			return
 		}
 		now = time.Now()
 		time.Sleep(10 * time.Second)
 	}
-	log.Infof("I should submit the transaction now!")
 
 	tasm := &validatorproto.TopicAggregatedSignaturesMessage{
-		Hash:      hexHash,
+		Hash:      hash,
 		EthTxHash: "0x12345",
 	}
 
-	_, err = proto.Marshal(tasm) // topicMessageBytes
+	topicMessageBytes, err := proto.Marshal(tasm)
 	if err != nil {
 		log.Error("Could not marshal Topic Aggregated Signatures Message [%s] - [%s]", tasm, err)
 	}
 
-	log.Infof("I should send the message! - [%v]", tasm)
-	//_, err = cmh.hederaNodeClient.SubmitTopicConsensusMessage(cmh.topicID, topicMessageBytes)
-	//if err != nil {
-	//	log.Error("Could not Submit Topic Consensus Message [%s] - [%s]", topicMessageBytes, err)
-	//}
+	_, err = cmh.hederaNodeClient.SubmitTopicConsensusMessage(cmh.topicID, topicMessageBytes)
+	if err != nil {
+		log.Error("Could not submit Topic Consensus Message [%s] - [%s]", topicMessageBytes, err)
+	}
+}
 
-	return nil
+func (cmh ConsensusMessageHandler) enoughSignaturesCollected(txSignatures []message.TransactionMessage, transactionId string) bool {
+	requiredSigCount := len(cmh.operatorsEthAddresses)/2 + len(cmh.operatorsEthAddresses)%2
+	log.Infof("Required signatures: [%v]", requiredSigCount)
```

Comment:
this log is maybe a leftover for debug purposes only?
If you think that this info is meaningful (I do), you should put it in the log with tx context (below on line 181)